MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/FullyDrawnStartupActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.macrobenchmark.target
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.View
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.platform.LocalView
+import androidx.core.view.doOnPreDraw
+
+class FullyDrawnStartupActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            TextBlock("Compose Macrobenchmark Target")
+        }
+    }
+
+    @Composable
+    fun ReportFullyDrawn(
```

Comment:
We are having a conversation about making it easier to report `reportFullyDrawn()` from Compose. This is one way of doing it, but ideally we do it after the frames have been rendered with a `Chroreographer` callback. Also, with `Lazy*` composables, it a bit more work, because that has to start with a `Flow`'s emission.

Reply:
Where is that conversation happening, Rahul?

Macrobench expects reportFullyDrawn to happen *before or during* the frame it is associated with. If you delay it until after the choreographer callback, macrobench will report it as associating with next frame.

While platform reports it immediately when it's called, we're treating this as a bug (https://b.corp.google.com/issues/203557137 - internal bug) - as that way it doesn't account for renderthread costs.